server/discover.go

```diff
@@ -352,6 +352,10 @@ func (s *Server) entryToRequest(entry *set.Entry, transformer put.PathTransforme
 		r.Meta[put.MetaKeySymlink] = entry.Dest
 	}
 
+	for i, v := range given.Metadata {
```

Comment:
This still needs updating? The i loop variable, and if it's at all possible for given.Metadata to have bypassed the client-side validity check, the check should happen again here.